### DISCUSSION TITLE
CAPTCHA support

### DISCUSSION THREAD

**Discussion by pbiering** [sni/thruk#1347] [Ideas]
anyone interested adding CAPTCHA support to Thruk login window?

I've implemented this in Perl already for https://github.com/pbiering/letterbox-sensor-web
in https://github.com/pbiering/letterbox-sensor-web/blob/master/ttn-letterbox-userauth.pm and related config options documented here https://github.com/pbiering/letterbox-sensor-web/blob/master/ttn-letterbox.conf

for at least Google, hCaptcha & FriendlyCaptcha

If one provide me some hints where to hook best into Thruk's authentication layer I can try my best to contribute...if it is not easier by the maintainer to takeover example code...

**Comment by sni** [comment]
may i ask for the reason? Usually Thruk runs on internal networks and i never saw any captcha on such applications. Is this intended for a public facing setup?

**Comment by pbiering** [comment]
> Is this intended for a public facing setup?

Indeed, at least for my installation on personal VPS (currently ACL protected), but also other external hosted monitoring probes can be considered.

**Comment by sni** [comment]
well, personally i don't like captchas and nowadays AI is much better in solving captchas then humans, so to be honest, i have little interest in adding or maintaining this.
Usually the audience is a limited group, so client certificates or ACLs are probably a better and less annoying solution.
